test/e2e/config/azure-dev.yaml

```diff
@@ -92,7 +92,7 @@ providers:
       replacements:
       - old: "imagePullPolicy: Always"
         new: "imagePullPolicy: IfNotPresent"
-    - name: v1.1.99 # next; use manifest from source files
+    - name: v1.2.99 # next; use manifest from source files
```

Comment:
this is a fake CAPZ version, not related to the CAPI version

should actually be v1.4.99 now if we're trying to represent a future patch of the current minor (it's completely arbitrary so it doesn't matter, just needs to be a valid semver so clusterctl doesn't complain)

Reply:
this still needs to be fixed @mboersma

Reply:
When updated to 1.4.99 these test failures resulted:

- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2431/pull-cluster-api-provider-azure-e2e/1550212646572134400

I reverted to 1.2.99, there clearly seems to be some significance w/ respect to the way that the cluster-api `metadata.yaml` spec processes this as input.

Reply:
After further consideration, I think we should do this in a separate PR:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2497

I've advocated that we actually use a "vNext" approach given that that's what main branch is meant to be testing. We can discuss that in the above PR thread.